src/cedalion/sim/synthetic_artifact.py

```diff
@@ -0,0 +1,365 @@
+    ]
+    return windows
+
+
+def calculate_amplitudes(windows):
+    """Calculates the amplitude (max-min) for each window."""
+
+    windows = windows.pint.dequantify()
+
+    amplitudes = [np.max(window) - np.min(window) for window in windows]
+    return np.array(amplitudes)
+
+
+def compute_alpha(ts: cdt.NDTimeSeries, window_size, step_size):
+    """Computes the Alpha parameter using the sliding window approach.
+
+    Args:
+        ts: fNIRS timeseries data (i.e. amp or OD).
+        window_size: Size of the window in s.
+        step_size: Step size in s.
+
+    Returns:
+        Median amplitude of the windows.
+    """
+
+    windows = sliding_window(ts, window_size, step_size)
+    amplitudes = calculate_amplitudes(windows)
+    return np.median(amplitudes)
+
+
+def add_artifact_direct(
+    ts: cdt.NDTimeSeries, timing: tuple[float, float], artifact_func, scale: float = 1.0
+):
+    """Add a single artifact to a timeseries with direct scaling (mainly for testing).
+
+    Args:
+        ts: timeseries data
+        timing: Tuple with onset time, duration of the artifact.
+        artifact_func: Artifact function. Artifact function must take args (time).
+        scale: scale for artifact.
+
+    Returns:
+        Timeseries data with added artifact.
+    """
+
+    ts_copy = ts.copy()
+    unit = ts_copy.pint.units if ts_copy.pint.units else 1
+
+    artifact = artifact_func(ts_copy.time, timing[0], timing[1])
+    ts_copy += artifact * scale * unit
+    return ts_copy
+
+
+def add_artifacts(
+    ts: cdt.NDTimeSeries,
+    timing: pd.DataFrame,
+    artifacts: dict[str, ArtifactFunction],
+    mode: str = "auto",
+    scale: float = 1.0,  # FIXME combine with mode
+) -> cdt.NDTimeSeries:
+    """Add scaled artifacts to timeseries data.
+
+    Supports timeseries with channel and either wavelength or chromophore dimension.
+    Currently assumes that artifacts affect both wavelengths/chromophores equally.
+
+    Args:
+        ts: fNIRS timeseries data (i.e. amp or OD).
+        timing: DataFrame with event timing data (columns onset_time, duration,
+            trial_type, value, channel).
+        artifacts: Dictionary of artifact functions. Artifact functions must take args
+            (time, onset_time, duration). Keys correspond to the trial_type in the
+            timing DataFrame.
+        mode: 'auto' or 'manual'. If 'auto', artifacts are scaled using the alpha
+            parameter (median of median of sliding windows). If 'manual', artifacts
+            are scaled only by the scale parameter.
+        scale: scaling parameter for artifacts
+
+    Returns:
+        Amplitude data with added artifacts.
+    """
+
+    ts_copy = ts.copy()
+    unit = ts_copy.pint.units if ts_copy.pint.units else 1
+
+    time_start = ts_copy["time"][0].item()
+    time_end = ts_copy["time"][-1].item()
+
+    # set parameters for compute_alpha
+    window_size = ts_copy.time.size // 100 # FIXME window size depends on len(ts_copy)?
```

Comment:
I'm not sure if this is desirable. This way, the window size depends on the length of the dataset.

Reply:
Ok, do you think window size should be an argument or hardcoded? The study I got this technique from used 120s, overlapping, but I'm not sure if this is generally applicable.

Reply:
I'm also not sure about the general applicability. Hence, let's make it a configurable parameter for which 120s, overlapping, should be a good default.